arcgis-ios-sdk-samples/Edit data/Edit with branch versioning/EditWithBranchVersioningViewController.swift

```diff
@@ -0,0 +1,542 @@
+            undoLocalEdits(geodatabase: geodatabase) {
+                geodatabase.switchVersion(withName: branchVersionName) { [weak self] error in
+                    guard let self = self else { return }
+                    if let error = error {
+                        self.presentAlert(error: error)
+                    } else {
+                        // Reload feature table with the new version.
+                        self.featureLayer.featureTable?.load()
+                        self.currentVersionName = branchVersionName
+                    }
+                }
+            }
+        } else {
+            // Apply local edits when switching to a user created branch.
+            applyLocalEdits(geodatabase: geodatabase) {
+                geodatabase.switchVersion(withName: branchVersionName) { [weak self] error in
+                    guard let self = self else { return }
+                    if let error = error {
+                        self.presentAlert(error: error)
+                    } else {
+                        // Reload feature table with the new version.
+                        self.featureLayer.featureTable?.load()
+                        self.currentVersionName = branchVersionName
+                    }
+                }
+            }
+        }
+    }
+    
+    /// Apply local edits to the geodatabase.
+    ///
+    /// - Parameters:
+    ///   - geodatabase: The geodatabase to apply edits.
+    ///   - completion: An optional closure to execute after edits are applied.
+    func applyLocalEdits(geodatabase: AGSServiceGeodatabase, completion: (() -> Void)? = nil) {
+        if geodatabase.hasLocalEdits() {
+            SVProgressHUD.show(withStatus: "Applying local edits…")
+            geodatabase.applyEdits { _, _ in
+                SVProgressHUD.dismiss()
+                completion?()
+            }
+        } else {
+            completion?()
+        }
+    }
+    
+    /// Undo local edits on the geodatabase.
+    ///
+    /// - Parameters:
+    ///   - geodatabase: The geodatabase to discard edits.
+    ///   - completion: An optional closure to execute after edits are undone.
+    func undoLocalEdits(geodatabase: AGSServiceGeodatabase, completion: (() -> Void)? = nil) {
+        if geodatabase.hasLocalEdits() {
+            SVProgressHUD.show(withStatus: "Discarding local edits…")
+            geodatabase.undoLocalEdits { _ in
+                SVProgressHUD.dismiss()
+                completion?()
+            }
+        } else {
+            completion?()
+        }
+    }
+    
+    // MARK: Actions
+    
+    @IBAction func createBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        // Clear selection before creating a new branch version.
+        clearSelection()
+        mapView.callout.dismiss()
+        chooseVersionAccessPermission(sender) { permission in
+            self.createBranchAlert(permission: permission) { [weak self] parameters in
```

Comment:
I tried to remove the weak capture here but it seems to cause the VC not to release. 🤔 

When the "branch details" alert is dismissed, since it calls an escaping closure, I'm not sure the alert can recycle everything. I'll leave it to the next round of review.

Reply:
Which view controller? The sample view controller or the alert controller?

Reply:
The sample's `EditWithBranchVersioningViewController`. Without the weak capture it seems not to deinit on exit.